programs/x509/cert_write.c

```diff
@@ -235,6 +235,41 @@ int write_certificate( mbedtls_x509write_cert *crt, const char *output_file,
     return( 0 );
 }
 
+/*
+ * Convert the input "in_buff" string to a raw byte array "out_buff". The amount
+ * of converted data is returned on "written_data".
+ */
+static int parse_serial( const char* in_buff, size_t in_buff_len,
+                    unsigned char* out_buff, size_t out_buff_len,
+                    size_t *written_data )
+{
+    char c;
+    unsigned char val;
+    int i;
+
+    if( out_buff_len < in_buff_len )
```

Comment:
That (and the code below) doesn't seem correct, as takes two bytes of hex input to encode one byte of binary output. Fortunately we have a function for that that's available to programs.

Reply:
I wrote a comment about this on slack: basically I misinterpreted the relation between input and output buffers, but I got it now and will be fixed.